### PR TITLE
Add monitor and failover support to DNSMadeEasy module

### PR DESCRIPTION
**Pull request by ericworkman**
##### SUMMARY
Extends the DNSMadeEasy module to allow for monitoring and failover configuration for A records.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
DNSMadeEasy

##### ANSIBLE VERSION
```
ansible 2.4.0
```

##### ADDITIONAL INFORMATION
I've manually tested failover and monitor configuration in all the permutations I could think of using the [DNSMadeEasy sandbox](http://sandbox.dnsmadeeasy.com).

- Existing modules options still work
- Required groups of parameters are specified according to what I could work out from [DNSMadeEasy's documentation](https://www.dnsmadeeasy.com/pdf/API-Docv2.pdf) and from poking at the API
- Applying both monitoring and failover work or fail with helpful error messages
- Removal of monitoring and failover work
- The module reports changed status appropriately if either the monitor "record" or the record change

### CODE REVIEW

File: lib/ansible/modules/network/dnsmadeeasy.py
Comment (gundalow) [inline lib/ansible/modules/network/dnsmadeeasy.py]: `version_added: "2.4"`

File: lib/ansible/modules/network/dnsmadeeasy.py
Comment (gundalow) [inline lib/ansible/modules/network/dnsmadeeasy.py]: Descriptions should be full sentences, and end with a full stop.

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by ericworkman** [comment]
@gundalow Thanks for the review! How's this look?

**Comment by gundalow** [comment]
Hi,
A number of files have been git moved from `modules/network` to `modules/net_tools`, which has effected this PR. Please rebase this PR to pull in that change.

**Comment by gundalow** [comment]
@briceburg I've given this PR the once over and it looks good, as the module maintainer we'd like to get your approval before it gets merged (pending on rebase)

**Comment by ericworkman** [comment]
I've rebased this PR. Thank you @gundalow !

**Review by gundalow — APPROVED** [review]

**Comment by briceburg** [comment]
Thanks @gundalow. I say shipit

**Comment by gundalow** [comment]
Thank you all for the PR and the reviews. This has been merged into `devel` and will be released in Ansible 2.4, as we don't backport new features.

**Pull request merged**
